domain-server/src/DomainGatekeeper.cpp

```diff
@@ -1061,3 +1207,91 @@ Node::LocalID DomainGatekeeper::findOrCreateLocalID(const QUuid& uuid) {
     _localIDs.insert(newLocalID);
     return newLocalID;
 }
+
+
+bool DomainGatekeeper::domainHasLogin() {
+    // The domain may have its own users and groups in a WordPress site.
+    return _server->_settingsManager.valueForKeyPath(AUTHENTICATION_ENABLE_OAUTH2).toBool()
+        && !_server->_settingsManager.valueForKeyPath(AUTHENTICATION_OAUTH2_URL_PATH).toString().isEmpty()
+        && !_server->_settingsManager.valueForKeyPath(AUTHENTICATION_WORDPRESS_URL_BASE).toString().isEmpty()
+        && !_server->_settingsManager.valueForKeyPath(AUTHENTICATION_PLUGIN_CLIENT_ID).toString().isEmpty();
+}
+
+void DomainGatekeeper::requestDomainUser(const QString& username, const QString& accessToken, const QString& refreshToken) {
```

Comment:
There doesn't seem to be any use made of refresh tokens, and no expiration date set for access tokens.  I'm thinking the access token expiration date would need to be included in processConnectRequestPacket (unless there's some way to query for its expiration date?), otherwise this may prompt the user for a login more often than expected.

Reply:
Handling of token expiry and refresh, unfortunately, hasn't made it into this cut of the code. It will be handled in due course.